solc/CommandLineInterface.cpp

```diff
@@ -157,20 +158,26 @@ void CommandLineInterface::handleBinary(string const& _contract)
 {
 	solAssert(m_options.input.mode == InputMode::Compiler || m_options.input.mode == InputMode::CompilerWithASTImport, "");
 
-	if (m_options.compiler.outputs.binary)
+	solAssert(m_options.compiler.outputs.has_value(), "");
```

Comment:
Messages in our assert macros are now optional thanks to #12019.
```suggestion
	solAssert(m_options.compiler.outputs.has_value());
```